### DISCUSSION TITLE
Where can I find documentation about the `renderer response`?

### DISCUSSION THREAD

**Discussion by vabatta** [nitrojs/nitro#1853] [Q&A]
I would like to extend the nitro server runtime with a plugin to process HTML responses with a custom templating engine. 
By looking at the docs, I have the feeling that the `renderer response` would be the right place to start building and hooking such a task.

**Comment by stafyniaksacha** [comment]
You can just create an event handler, render your template and return the result.

As an example, you can take a look at AnalogJS which use nitro to perform SSR on angular apps. The handler is pretty straightforward: https://github.com/analogjs/analog/blob/main/packages/vite-plugin-nitro/src/lib/runtime/renderer.js
Here is how it's loaded: https://github.com/analogjs/analog/blob/main/packages/vite-plugin-nitro/src/lib/vite-plugin-nitro.ts#L144

A more complex example is nuxt, here is the renderer definition: https://github.com/nuxt/nuxt/blob/main/packages/nuxt/src/core/runtime/nitro/renderer.ts#L201
and here is the binding with nitro: https://github.com/nuxt/nuxt/blob/main/packages/nuxt/src/core/nitro.ts#L89
